pkg/openslo/v1/objects.go

```diff
@@ -49,3 +53,72 @@ func (a *Label) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+func validationRulesAPIVersion[T openslo.Object](
+	getter func(T) openslo.Version,
+) govy.PropertyRules[openslo.Version, T] {
+	return govy.For(getter).
+		WithName("apiVersion").
+		Required().
+		Rules(rules.EQ(APIVersion))
+}
+
+func validationRulesKind[T openslo.Object](
+	getter func(T) openslo.Kind, kind openslo.Kind,
+) govy.PropertyRules[openslo.Kind, T] {
+	return govy.For(getter).
+		WithName("kind").
+		Required().
+		Rules(rules.EQ(kind))
+}
+
+func validationRulesMetadata[T openslo.Object](getter func(T) Metadata) govy.PropertyRules[Metadata, T] {
+	return govy.For(getter).
+		WithName("metadata").
+		Include(
+			govy.New(
+				govy.For(func(m Metadata) string { return m.Name }).
+					WithName("name").
+					Required().
+					Rules(rules.StringDNSLabel()),
```

Comment:
I see this is only validation for strict rules. Do we want to add strict level switch in future PRs? I have been thinking about it for a bit and it feels like Specification should be simple and unambiguous, otherwise we will get to the situation when two vendors will claim they support OpenSLO spec, but both do it differently.

My suggest to remove all cases when specification can be expanded. Lmk what do you think and I will convert this comment into issue.

Reply:
I thought about that too, the idea I had was to define an enum that the `Validate` method will accept, wdyt about that?

Reply:
I know, this was my idea too at beginning.

But one of missions of OpenSLO is to be able to move between different vendors easily using the same code defining SLOs, if one vendor will accept names that are 63 characters and other will accept 255 it won't be possible without any additional actions. This is why I suggest that we should use only one so there is clarity whether someone is compatible with OpenSLO or not no in between states.

Reply:
I think you can move forward with current solution and I will write an issue describing the problem